Detectors/EMCAL/simulation/src/DMAOutputStream.cxx

```diff
@@ -41,30 +53,30 @@ void DMAOutputStream::writeData(RawHeader header, gsl::span<char> buffer)
   // from the maximum possible payload size
   constexpr int MAXNWORDS = PAGESIZE - sizeof(header) - sizeof(uint32_t);
   bool writeNext = true;
-  int pagecounter = 0, currentindex = 0;
+  int pagecounter = header.pageCnt, currentindex = 0;
   while (writeNext) {
     int sizeRemain = buffer.size() - currentindex;
     int nwordspage = MAXNWORDS;
     if (sizeRemain < MAXNWORDS) {
       // Last page
       nwordspage = sizeRemain;
       writeNext = false;
-      header.stop = true;
     }
-    header.packetCounter = pagecounter;
+    header.pageCnt = pagecounter;
     header.memorySize = nwordspage + sizeof(RawHeader);
     header.offsetToNext = 8192;
```

Comment:
Will you always write fixed size pages?

Reply:
For the moment this is the status that we have a fixed page size of 8kB, if WP6 changes this I will adapt the page size handling.